src/app/shared/date/timepicker.component.ts

```diff
@@ -19,40 +15,94 @@ export class TimepickerComponent {
   @Input() date: Date;
   @Output() onTimeChange = new EventEmitter<Date>();
   @Input() changed: boolean;
-  hourOptions: string[] = new Array(24).fill('').map((x, i) => i < 10 ? '0' + i : '' + i);
-  minuteOptions = ['00', '30'];
-
-  get hour(): string {
-    if (!this.date) {
-      return '00';
-    } else {
-      let date = new Date(this.date.valueOf());
-      if (date.getMinutes() <= 30) {
-        return date.getHours() < 10 ? '0' + date.getHours() : '' + date.getHours();
-      } else {
-        return date.getHours() + 1 <= 23 ? '' + (date.getHours() + 1) : '00';
-      }
-    }
+
+  localTimezone = new Date().toString().match(/(\([A-Za-z\s].*\))/)[1];
+  options: string[] = [
```

Comment:
Trying to think of a clever way to generate this.  Lots of crazy math solutions out there, but the one i like best just uses Date...

```
let times = [];
let day = new Date(1970, 0, 1, 0, 0, 0, 0);
while (day.getDate() === 1) {
  times.push(day.toTimeString());
  day.setMinutes(day.getMinutes() + 30);
}
```